Sources/FoundationEssentials/Data/DataProtocol.swift

```diff
@@ -10,11 +10,11 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if os(macOS) || os(iOS)
+#if canImport(Darwin)
 import Darwin
-#elseif os(Linux)
+#elseif canImport(Glibc)
 import Glibc
-#elseif os(Windows)
+#elseif canImport(ucrt)
 import ucrt
 #endif
```

Comment:
Would you mind adding support for Musl and WASI while we're at it?

```suggestion
#elseif canImport(Musl)
import Musl
#elseif canImport(ucrt)
import ucrt
#elseif canImport(WASILibc)
import WASILibc
#endif
```